src/UdpDataProtocol.cpp

```diff
@@ -507,18 +525,17 @@ void UdpDataProtocol::run()
         if (gVerboseFlag) std::cout << "    UdpDataProtocol:run" << mRunMode << " before !UdpSocket.hasPendingDatagrams()" << std::endl;
         std::cout << "Waiting for Peer..." << std::endl;
         // This blocks waiting for the first packet
-        while ( !UdpSocket.hasPendingDatagrams() ) {
+        while ( !datagramAvailable() ) {
             if (mStopped) { return; }
             QThread::msleep(100);
             if (gVerboseFlag) std::cout << "100ms  " << std::flush;
         }
-        int first_packet_size = UdpSocket.pendingDatagramSize();
         // The following line is the same as
-        int8_t* first_packet = new int8_t[first_packet_size];
+        int8_t* first_packet = new int8_t[full_redundant_packet_size];
         /// \todo fix this to avoid memory leaks
         // but avoids memory leaks
         //std::tr1::shared_ptr<int8_t> first_packet(new int8_t[first_packet_size]);
-        receivePacket( UdpSocket, reinterpret_cast<char*>(first_packet), first_packet_size);
+        receivePacket( reinterpret_cast<char*>(first_packet), full_redundant_packet_size);
```

Comment:
I thought the first packet is just the handshake. Why do we need full redundant packet size at this point? Actually those packet size names are not clear. But that's for another day.

Reply:
We don't. Technically we could just use the header size and let the rest of the packet go, but reading the header size would have meant adding an additional getter in the JackTrip class. And this way we can just allocate a single buffer for both the initial packet and all of the subsequent packets.

Reply:
(It's worth noting that the code that this replaces used to read the full first packet into the buffer as well, so this is no less efficient than the previous code. And the first packet isn't any smaller than the packets that follow it - there's never a header sent by itself without audio data.)